fe1-web/model/objects/WalletCryptographyHandler.ts

```diff
@@ -15,62 +30,134 @@ export class WalletCryptographyHandler {
+     In jest context provide a MOCK crypto.subtle library, otherwise provide window.crypto */
+  constructor(cryptography: object) {
+    this.cryptography = cryptography;
+  }
 
   /**
-   * this method adds an RSA encryption/decryption key,
-   * interacts with IndexedDB browser database.
+   * This functions verifies weather or not the wallet storage in IndexedDB database
+   * is initialised. If not it initialises the storage.
    */
-  public async initWalletStorage(): Promise <void> {
-    const key: WalletCryptoKey = await this.generateRSAKey();
-    await this.keyDatabase.putKey(key);
+  public async initWalletStorage(): Promise<void> {
+    /* verifies if wallet storage has already been initialised */
+    const tryPublicKey: CryptoKey = await this.getKeyFromDatabase(STRINGS.walletPublicKey);
+    const tryPrivateKey: CryptoKey = await this.getKeyFromDatabase(STRINGS.walletPrivateKey);
+    const walletIsNotInitialised: boolean = (tryPublicKey === undefined)
+      || (tryPrivateKey === undefined);
+
+    if (walletIsNotInitialised) {
+      await this.handleWalletInitialization();
+    }
+    console.log('Wallet storage ready');
   }
 
   /**
-   * encrypts the given ed25519 token with the RSA key stored in the indexedDB database
-   * @param token ed25519 key toUint8Array()
+   * encrypts the given ed25519 seed with the RSA key stored in the indexedDB database
+   * @param plaintext ed25519 seed toUint8Array()
    */
-  public async encrypt(token: Uint8Array): Promise<ArrayBuffer> {
-    const key = await this.getKeyFromDatabase(STRINGS.walletPublicKey);
-    const cypheredToken = await window.crypto.subtle.encrypt(this.algorithm, key, token);
-    return cypheredToken;
+  public async encrypt(plaintext: Uint8Array): Promise<ArrayBuffer> {
+    const key: CryptoKey = await this.getKeyFromDatabase(STRINGS.walletPublicKey);
+    if (key === undefined) {
+      throw Error('Error while retrieving encryption key from database: undefined');
+    }
+    // @ts-ignore
+    const ciphertext = await this.cryptography.subtle.encrypt(this.algorithm, key, plaintext);
+    return ciphertext;
   }
 
   /**
-   * decrypts the encrypted ed25519 token with the RSA key stored in the indexedDB database
-   * @param encryptedToken ed25519 encrypted token (ArrayBuffer)
+   * decrypts the encrypted ed25519 seed with the RSA key stored in the indexedDB database
+   * @param ciphertext ed25519 encrypted seed (ArrayBuffer)
    */
-  public async decrypt(encryptedToken: ArrayBuffer): Promise<ArrayBuffer> {
+  public async decrypt(ciphertext: ArrayBuffer): Promise<ArrayBuffer> {
     const key = await this.getKeyFromDatabase(STRINGS.walletPrivateKey);
-    const plaintextToken = await window.crypto.subtle.decrypt(this.algorithm, key, encryptedToken);
-    return plaintextToken;
+    if (key === undefined) {
+      throw Error('Error while retrieving decryption key from database: undefined');
+    }
+    // @ts-ignore
```

Comment:
no need for `ts-ignore` if you define `cryptography` with the right type (`Crypto`)